x-pack/plugins/apm/public/components/shared/TraceLink.tsx

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { TransactionV2 } from '../../../typings/Transaction';
+import { ITransactionGroup } from '../../../typings/TransactionGroup';
+import { KibanaLink, legacyEncodeURIComponent } from '../../utils/url';
+
+interface TraceLinkProps {
+  transactionDoc?: TransactionV2;
+  transactionGroup?: ITransactionGroup | { [key: string]: any };
+}
+
+/**
+ * Return the path and query used to build a trace link,
+ * given either a v2 Transaction or a Transaction Group
+ */
+export function getLinkProps({
+  transactionDoc,
+  transactionGroup = {}
+}: TraceLinkProps) {
+  let {
+    serviceName,
+    transactionType,
+    traceId,
+    id: transactionId,
+    name
+  } = transactionGroup;
+
+  if (transactionDoc) {
+    serviceName = transactionDoc.context.service.name;
+    transactionType = transactionDoc.transaction.type;
+    traceId = transactionDoc.trace.id;
+    transactionId = transactionDoc.transaction.id;
+    name = transactionDoc.transaction.name;
+  }
+
+  if (!serviceName || !transactionType || !name) {
```

Comment:
I like this a lot. Will update.

Reply:
As I went to change this, I had a thought: what do you think about this instead (assuming the size for the query is always 1 here so it shouldn't affect query perf)...
```js
export interface ITransactionGroup {
  name: string;
  sample: Transaction;
  p95: number;
  averageResponseTime: number;
  transactionsPerMinute: number;
  impact: number;  
}
```
Then we can re-use everything we already know about `Transaction` types, and the `TraceLink` doesn't have to deal with multiple data structures bc it will always just get a `Transaction`...

Reply:
LGTM!